File: code/widgets/org.eclipse.scout.jswidgets.ui.html/src/main/js/formfield/StatusSeverityLookupCall.js
/*
 * Copyright (c) 2017 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Distribution License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/org/documents/edl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 */
import {StaticLookupCall, Status} from '@eclipse-scout/core';

export default class StatusSeverityLookupCall extends StaticLookupCall {

constructor() {
  super();
}


_data() {
  return StatusSeverityLookupCall.DATA;
}

static DATA = [
  [Status.Severity.INFO, 'info'],
  [Status.Severity.WARNING, 'warning'],
  [Status.Severity.ERROR, 'error']
];
}

File: modules/common/examples/widgets/org.eclipse.scout.widgets.client/src/main/java/org/eclipse/scout/widgets/client/ui/template/formfield/StatusSeverityLookupCall.java
/*
 * Copyright (c) 2018 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 */
package org.eclipse.scout.widgets.client.ui.template.formfield;

import java.util.ArrayList;
import java.util.List;

import org.eclipse.scout.rt.platform.classid.ClassId;
import org.eclipse.scout.rt.platform.status.IStatus;
import org.eclipse.scout.rt.shared.services.lookup.ILookupRow;
import org.eclipse.scout.rt.shared.services.lookup.LocalLookupCall;
import org.eclipse.scout.rt.shared.services.lookup.LookupRow;

@ClassId("cfe2058f-bc4b-411e-8ff6-61eb8e7714e3")
public class StatusSeverityLookupCall extends LocalLookupCall<Integer> {

  private static final long serialVersionUID = 1L;

  @Override
  protected List<? extends ILookupRow<Integer>> execCreateLookupRows() {
    List<ILookupRow<Integer>> rows = new ArrayList<>();
    rows.add(new LookupRow<>(IStatus.OK, "ok"));
    rows.add(new LookupRow<>(IStatus.INFO, "info"));
    rows.add(new LookupRow<>(IStatus.WARNING, "warning"));
    rows.add(new LookupRow<>(IStatus.ERROR, "error"));
    return rows;
  }
}
